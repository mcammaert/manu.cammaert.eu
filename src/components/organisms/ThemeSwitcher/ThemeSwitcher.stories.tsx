import React from 'react';

import { boolean, number, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import ThemeSwitcher from './ThemeSwitcher';

storiesOf('ThemeSwitcher', module).add('ThemeSwitcher', () => {
  return <ThemeSwitcher />;
});
