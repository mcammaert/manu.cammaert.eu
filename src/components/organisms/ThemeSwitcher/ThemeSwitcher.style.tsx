import styled from 'styled-components';
import { modularScale } from 'polished';

import { NavigationItem } from 'components/atoms/NavigationItem';

export const ThemeSwitcher = styled.div`
  margin-top: ${modularScale(3)};
`;

export const Label = styled.p`
  font-size: ${modularScale(0)};
  font-weight: 400;
  line-height: ${modularScale(1)};
  color: ${({ theme }) => theme.colors.copy};
`;

export const Buttons = styled.div`
  display: flex;
`;

export const Button = styled(NavigationItem)`
  margin-right: ${modularScale(0)};
  flex-grow: 1;
  &:last-of-type {
    margin-right: 0;
  }
`;

export default { ThemeSwitcher, Label, Buttons, Button };
