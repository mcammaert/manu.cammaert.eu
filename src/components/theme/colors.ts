export const colors = {
  primary90: '#0c2d48',
  primary80: '#0b395d',
  primary70: '#0b4573',
  primary60: '#0d5189',
  primary50: '#145da0',
  primary40: '#367bb1',
  primary30: '#5c99c1',
  primary20: '#85b7d0',
  primary10: '#b1d4e0',

  secondary90: '#ca3f36',
  secondary80: '#d35239',
  secondary70: '#db633e',
  secondary60: '#e37443',
  secondary50: '#eb844a',
  secondary40: '#f79b77',
  secondary30: '#fdb5a1',
  secondary20: '#fdd0c8',
  secondary10: '#f7eceb',

  accent1: '#d45e95',
  accent2: '#b290d4',
  accent3: '#97b8e6',

  // 10 shades of grey
  black: '#1E2025',
  charcoal: '#262830',
  shadow: '#2E313A',
  lead: '#585965',
  steel: '#787C8A',
  gray: '#989DAC',
  smoke: '#BEC2C9',
  fog: '#EDEEF0',
  white: '#F6F7F9',
  snow: '#FCFCFC',

  // slate
  slateLight: '#3279AD',
  slate: '#4C6E89',
  slateDark: '#455C77',

  // teal
  teal: '#3D829C',
  tealDark: '#36768B',

  // laguna
  lagunaLight: '#4FB8F3',
  laguna: '#57A7E3',
  lagunaDark: '#3680DC',

  // warm colors
  red: '#F60C5B',
  orange: '#F86C5C',
  sunflower: '#EDC172',
  lemon: '#FFE362',

  // purples
  grape: '#374278',
  plum: '#725FA4',
  lavender: '#936FBA',
  purple: '#C163C6',

  // greens
  forest: '#50C04E',
  lawn: '#86DE6A',
  aquamarine: '#14DAB5',
  emerald: '#0DCACE',
  turquoise: '#57D5D6',

  // browns
  chocolate: '#3C322D',
  barnwood: '#5E5E5F',
};

export default colors;
