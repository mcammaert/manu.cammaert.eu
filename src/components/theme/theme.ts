import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body {
    background-color: #ccc;
  }
`;

const theme = {
  main: 'mediumseagreen',
};

export default theme;
