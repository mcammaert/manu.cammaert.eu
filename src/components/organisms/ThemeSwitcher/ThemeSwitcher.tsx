import React from 'react';

import { useCookies } from 'react-cookie';

import { ThemeSwitcherProps } from './ThemeSwitcher.types';

import S from './ThemeSwitcher.style';

const ThemeSwitcher: React.FC<ThemeSwitcherProps> = () => {
  const [{ theme }, setCookies] = useCookies(['theme']);
  const darkModeSupport = window.matchMedia('(prefers-color-scheme)').media !== 'not all';

  const toggleLightThemeHandler = () => {
    setCookies('theme', 'light');
  };

  const toggleAutoThemeHandler = () => {
    setCookies('theme', 'auto');
  };

  const toggleDarkThemeHandler = () => {
    setCookies('theme', 'dark');
  };

  return (
    <S.ThemeSwitcher>
      <S.Label>Weergave</S.Label>
      <S.Buttons>
        <S.Button onClick={toggleLightThemeHandler} disabled={theme === 'light'}>
          Licht
        </S.Button>
        <S.Button onClick={toggleDarkThemeHandler} disabled={theme === 'dark'}>
          Donker
        </S.Button>
        {darkModeSupport && (
          <S.Button onClick={toggleAutoThemeHandler} disabled={theme === 'auto'}>
            Automatisch
          </S.Button>
        )}
      </S.Buttons>
    </S.ThemeSwitcher>
  );
};

export default ThemeSwitcher;
