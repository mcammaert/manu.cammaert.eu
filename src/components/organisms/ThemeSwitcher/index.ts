import * as S from './ThemeSwitcher.style';

export { default as ThemeSwitcher } from './ThemeSwitcher';
export * from './ThemeSwitcher.types';
export { S };
