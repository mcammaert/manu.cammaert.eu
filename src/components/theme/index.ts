import { createGlobalStyle } from 'styled-components';
import 'normalize.css';
import { colors } from './colors';

export interface GlobalStyleProps {
  readonly light: boolean;
}

// const backgroundColor = (location: string) => (location === '/' ? colors.oxfordBlue : colors.fuzzyWuzzy);

export const GlobalStyle = createGlobalStyle<GlobalStyleProps>`
  body {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    font-family: 'Nunito Sans', sans-serif;
    width: 100vw;
    background: ${({ light }) => (light ? colors.white : colors.black)};
    background-attachment: fixed;    
    color: ${({ light }) => (light ? colors.black : colors.snow)};
    transition: all 800ms;
    a {
      color: ${({ theme }) => theme.colors.secondary};
    }
  }
`;

export const lightTheme = {
  margin: 15,
  colors: {
    primary: colors.charcoal,
    primaryInvert: colors.snow,
    secondary: colors.primary60,
    secondaryInvert: colors.white,
    tertiary: colors.secondary80,
    tertiaryInvert: colors.white,
    copy: colors.charcoal,
    copyInvert: colors.white,
  },
};

export const darkTheme = {
  margin: 15,
  colors: {
    primary: colors.snow,
    primaryInvert: colors.black,
    secondary: colors.primary30,
    secondaryInvert: colors.primary90,
    tertiary: colors.secondary30,
    tertiaryInvert: colors.secondary80,
    copy: colors.white,
    copyInvert: colors.charcoal,
  },
};
