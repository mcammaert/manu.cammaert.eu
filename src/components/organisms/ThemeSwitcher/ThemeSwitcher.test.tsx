import React from 'react';

import { cleanup } from '@testing-library/react';
import 'jest-styled-components';

import { renderComponent } from 'testing/react-testing-library';
import { lightTheme } from 'components/theme';

import ThemeSwitcher from './ThemeSwitcher';

afterEach(cleanup);

window.matchMedia = jest.fn().mockImplementation(query => {
  return {
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  };
});

describe('ThemeSwitcher', () => {
  describe('default props', () => {
    it('should match the snapshot', () => {
      const component = <ThemeSwitcher />;
      const { container } = renderComponent({ theme: lightTheme, component });
      expect(container).toMatchSnapshot();
    });
  });
});
