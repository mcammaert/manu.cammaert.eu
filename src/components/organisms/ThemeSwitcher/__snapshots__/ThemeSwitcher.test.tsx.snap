// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitcher default props should match the snapshot 1`] = `
.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1em;
  font-weight: 400;
  padding: 0.5627813555039173em 0.7501875468867217em;
  color: #F6F7F9;
  background: #0d5189;
  cursor: pointer;
  border: none;
  border-radius: 6px;
  box-shadow: 0 1px 18px rgba(30,32,37,0.08);
  opacity: 0.8;
  -webkit-transition: all 200ms;
  transition: all 200ms;
  -webkit-appearance: initial;
  padding: 1em;
  font-size: 1em;
  color: #262830;
  background: #FCFCFC;
}

.c3:hover:enabled {
  opacity: 1;
  cursor: pointer;
  box-shadow: 0 1px 18px rgba(30,32,37,0.25);
}

.c3:active {
  color: #0d5189;
  background: #F6F7F9;
}

.c3:disabled {
  opacity: 0.75;
  cursor: default;
  color: #F6F7F9;
  background: #0d5189;
}

.c3:active {
  color: #F6F7F9;
  background: #0d5189;
}

.c3:disabled {
  color: #F6F7F9;
  background: #0d5189;
}

.c0 {
  margin-top: 2.3685930369999997em;
}

.c1 {
  font-size: 1em;
  font-weight: 400;
  line-height: 1.333em;
  color: #262830;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  margin-right: 1em;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4:last-of-type {
  margin-right: 0;
}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
    >
      Weergave
    </p>
    <div
      class="c2"
    >
      <button
        class="c3 c4"
      >
        Licht
      </button>
      <button
        class="c3 c4"
      >
        Donker
      </button>
      <button
        class="c3 c4"
      >
        Automatisch
      </button>
    </div>
  </div>
</div>
`;
